go/libraries/doltcore/sqle/enginetest/branch_control_test.go

```diff
@@ -0,0 +1,687 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enginetest
+
+import (
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/enginetest"
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/mysql_db"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/src-d/go-errors.v1"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/branch_control"
+)
+
+// BranchControlTest is used to define a test using the branch control system. The root account is used with any queries
+// in the SetUpScript.
+type BranchControlTest struct {
+	Name        string
+	SetUpScript []string
+	Assertions  []BranchControlTestAssertion
+}
+
+// BranchControlTestAssertion is within a BranchControlTest to assert functionality.
+type BranchControlTestAssertion struct {
+	User           string
+	Host           string
+	Query          string
+	Expected       []sql.Row
+	ExpectedErr    *errors.Kind
+	ExpectedErrStr string
+}
+
+// BranchControlBlockTest are tests for quickly verifying that a command is blocked before the appropriate entry is
+// added to the "dolt_branch_control" table. The `TestUserSetUpScripts` are automatically run before every test,
+// therefore any set up here is essentially appended to `TestUserSetUpScripts`. In addition, the test user is
+// `testuser`@`localhost`.
+type BranchControlBlockTest struct {
+	Name        string
+	SetUpScript []string
+	Query       string
+	ExpectedErr *errors.Kind
+}
+
+// TestUserSetUpScripts creates a user named "testuser@localhost", and grants them privileges on all databases and
+// tables. In addition, creates a committed table named "test" with a single value, along with a mirror branch named
+// "other".
+var TestUserSetUpScripts = []string{
+	"DELETE FROM dolt_branch_control WHERE user = '%';",
```

Comment:
Technically it is since there's just that single row with `('%', '%', '%', 'write')` as the super user isn't an actual row, but I think it's a bit more clear what the intention is this way